shared/libraries/opcuatms/opcuatms/src/converters/core_types_converter.cpp

```diff
@@ -263,7 +263,9 @@ OpcUaObject<UA_Float> StructConverter<IFloat, UA_Float>::ToTmsType(const FloatPt
 template <>
 FloatPtr VariantConverter<IFloat>::ToDaqObject(const OpcUaVariant& variant, const ContextPtr& /*context*/)
 {
-    return variant.toDouble();
+    if (variant.isType<UA_Double>())
```

Comment:
I would maybe flip the condition.` if(variant-isType<UA_Float>()) toFolat(); else variant.toDouble();`